LethalAPI.Core/Events/Interfaces/IFirearmEvent.cs

```diff
@@ -0,0 +1,25 @@
+// -----------------------------------------------------------------------
+// <copyright file="IFirearmEvent.cs" company="LethalAPI Modding Community">
+// Copyright (c) LethalAPI Modding Community. All rights reserved.
+// Licensed under the GPL-3.0 license.
+// </copyright>
+// Taken from EXILED (https://github.com/Exiled-Team/EXILED)
+// Licensed under the CC BY SA 3 license. View it here:
+// https://github.com/Exiled-Team/EXILED/blob/master/LICENSE.md
+// Changes: Namespace adjustments, and potential removed properties.
+// -----------------------------------------------------------------------
+
+namespace LethalAPI.Core.Events.Interfaces;
+
+using UnityEngine;
+
+/// <summary>
+///     Event args used for all Firearme related events.
```

Comment:
Typo in Firearm

Also, we don't have firearms in LC?

Reply:
Doesn't this basically result in any GameObject / Item usage event?

Reply:
This is intended for the zap gun. The intention is to utilize a custom firearm wrappers for the firearm in the future.

I won't trigger anything unintentional now, and if we deem it unnecessary in the future, we can remove it.